return_smart_info.sh

```diff
@@ -2,18 +2,28 @@
 
 # Enumerate through all devices, which will pull in other things like
 # Device Mapper. Sometimes devices may not return full smart data if
-# are busy. This seems to work "okay" on a RHEL7 based laptop and
-# only results in the lines disappearing until available again.
+# are busy.
+# Tested on real bare metal with megaraid and nvme devices.
 
-for device in `ls /sys/block` ; do 
-    smartutil_output=$(mktemp --suffix=$i)
-    # Run the actual smartutil command
-    smartctl -i --attributes --log=selftest /dev/$device &>$smartutil_output
+CUT_CMD='cut -s -d ":" -f2-'
+SMART_CMD='smartctl -i --attributes --log=selftest'
 
-    serial=$(cat $smartutil_output |grep "Serial Number"|cut -c 37-)
-    capacity=$(cat $smartutil_output |grep "Total" |grep "Capacity" |cut -c 37-|sed 's/\ \[.*$//')
-    utilization=$(cat $smartutil_output |grep "Utilization" | cut -c 37-|sed 's/\ \[.*$//')
+get_serial () {
+    serial=$(cat $smartutil_output | grep -i "serial" | eval $CUT_CMD | sed "s/^[ \t]*//")
+}
 
+parse_smartutil_output_megaraid () {
+    # (rpittau) TODO: this is very basic, add more stuff
+    get_serial
```

Comment:
It might help to post this text someplace so we know what this looks like for development purposes.

Reply:
Adding some output samples with #11